src/utilities/util_dump.c

```diff
@@ -663,69 +670,110 @@ dump_record(WT_CURSOR *cursor, const char *key, bool reverse, bool search_near,
         suffix = "\n";
     }
 
-    while (ret == 0) {
-
-        /* A specific key is requested. */
-        if (key != NULL) {
-            current_key = key;
-            cursor->set_key(cursor, current_key);
-
-            ret = cursor->search_near(cursor, &exact);
-
-            if (ret != 0 && ret != WT_NOTFOUND)
-                return (util_cerr(cursor, "search_near", ret));
-
-            /*
-             * If a key has been found but search near is disabled and there is no exact match, the
-             * requested key is missing.
-             */
-            if (ret == 0 && !search_near && exact != 0)
-                ret = WT_NOTFOUND;
-
-            /* A key has been found. */
-            if (ret == 0) {
-                if (search_near && exact != 0) {
-                    /* Retrieve the key found by search_near. */
-                    if ((ret = cursor->get_key(cursor, &current_key)) != 0)
-                        return (util_cerr(cursor, "get_key", ret));
-                }
-                if ((ret = cursor->get_value(cursor, &value)) != 0)
-                    return (util_cerr(cursor, "get_value", ret));
-            }
+    if ((ret = cursor->get_key(cursor, &current_key)) != 0)
+        return (util_cerr(cursor, "get_key", ret));
+    if ((ret = cursor->get_value(cursor, &value)) != 0)
+        return (util_cerr(cursor, "get_value", ret));
+    if (fprintf(fp, "%s%s%s%s%s", prefix, current_key, infix, value, suffix) < 0)
+        return (util_err(session, EIO, NULL));
+    return (0);
+}
 
-        } else {
-            /* Continue parsing the file. */
-            ret = reverse ? cursor->prev(cursor) : cursor->next(cursor);
-            if (ret == 0) {
-                if ((ret = cursor->get_key(cursor, &current_key)) != 0)
-                    return (util_cerr(cursor, "get_key", ret));
-                if ((ret = cursor->get_value(cursor, &value)) != 0)
-                    return (util_cerr(cursor, "get_value", ret));
-            } else if (ret != WT_NOTFOUND)
-                return (util_cerr(cursor, (reverse ? "prev" : "next"), ret));
-        }
+/*
+ * dump_record --
+ *     Dump a single record, advance cursor to next/prev, along with JSON formatting if needed.
+ */
+static int
+dump_record(WT_CURSOR *cursor, const char *key, bool reverse, bool search_near, bool json, int window)
+{
+    WT_DECL_RET;
+    WT_SESSION *session;
+    int exact, n;
+    const char *current_key;
+    bool once;
+    int (*fwd)(WT_CURSOR *);
+    int (*bck)(WT_CURSOR *);
 
-        if (ret == 0) {
-            if (fprintf(fp, "%s%s%s%s%s%s", json && once ? "," : "", prefix, current_key, infix,
-                  value, suffix) < 0)
-                return (util_err(session, EIO, NULL));
+    session = cursor->session;
+    once = false;
+    exact = 0;
+
+    WT_ASSERT(session, key != NULL);
```

Comment:
`WT_ASSERT` expects a `WT_SESSION_IMPL *` as the first arg, sorry I missed that when we added this assertion.
```suggestion
    WT_ASSERT((WT_SESSION_IMPL *)session, key != NULL);
```

Reply:
fixed